scripts/cleanup_failed_registrations.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+TEMPDIR=`mktemp -d`
+trap "rm -rf $TEMPDIR" EXIT
```

Comment:
What's the purpose of this exactly? I see this pattern in several (but not all) of the bash files.